src/main/java/com/netgrif/application/engine/petrinet/service/PetriNetService.java

```diff
@@ -110,32 +112,40 @@ public class PetriNetService implements IPetriNetService {
     @Autowired
     private IUriService uriService;
 
-    private Map<ObjectId, PetriNet> cache = new HashMap<>();
+    @Autowired
+    private CacheManager cacheManager;
 
     protected Importer getImporter() {
         return importerProvider.get();
     }
 
     @Override
     public void evictCache() {
-        cache = new HashMap<>();
+        cacheManager.getCache("netsId").clear();
+        cacheManager.getCache("netsNewest").clear();
+        cacheManager.getCache("netsCache").clear();
+        cacheManager.getCache("netsIdentifier").clear();
+    }
+
+    @Override
+    public void evictCache(PetriNet net) {
+        cacheManager.getCache("netsId").evict(net.getStringId());
+        cacheManager.getCache("netsNewest").evict(net.getIdentifier());
+        cacheManager.getCache("netsCache").evict(net.getObjectId());
+        cacheManager.getCache("netsIdentifier").evict(net.getIdentifier() + net.getVersion().toString());
     }
 
     /**
      * Get read only Petri net.
      */
     @Override
+    @Cacheable("netsCache")
```

Comment:
Is it possible to apply the same cache to `List<PetriNet> get(Collection<ObjectId> petriNetIds)`  ? In general is it possible to reuse caches with single object argument to serve as caches for method with collection as argument?

Reply:
no , because cache ist basically hashMap, and it depends on key of method and on returning object of the method